### ISSUE TITLE
Proposal: Crediting "On Topic Comment Incentives" Only

### ISSUE THREAD

**Issue by pavlovcik** [ubiquity/ubiquibot#740]
This requires some experimentation but it would make sense to only offer credit to comment incentives that are on topic. 

When the comment incentives are being calculated, we should rely on ChatGPT to filter out any comments that are off topic.  

In order to save on network requests, we can consider the following:

1. Pass in every comment from users in a single array, and associate each node ID. 
2. We tell ChatGPT to consider the issue specification as the topic of conversation, and then ask it to go through every comment to see if any are off topic. 
3. Filter out off topic comments (referenced via comment node IDs)
4. We should be left with an array of "on topic" node IDs which can be passed into the comment calculation function.

**Comment by byteballet** [comment]
> I'm not sure how to handle long conversations that exceed the context window.

I think we can request several times in case the conversation is too long. We can trim the issue specification (this is called summarizing) to reduce the size of the request, split the comments into convenient sizes and we can send requests several times.
I don't see any necessity of sending request exactly once...

**Comment by pavlovcik** [comment]
- Multiple requests make our runtime significantly longer. 
- This makes our serverless runtime significantly longer. 
- That means we will pay significantly more money to run the bot. 

Instant runs (with caching if necessary) scales much better.

**Comment by byteballet** [comment]
Yes, so I am talking in only case the conversation is too long...
